src/main/kotlin/gg/essential/elementaPractice/gui/WhiteboardGUI.kt

```diff
@@ -0,0 +1,205 @@
+package gg.essential.elementaPractice.gui
+
+import gg.essential.elementa.ElementaVersion
+import gg.essential.elementa.WindowScreen
+import gg.essential.elementa.components.UIBlock
+import gg.essential.elementa.components.UIRoundedRectangle
+import gg.essential.elementa.components.UIText
+import gg.essential.elementa.constraints.CenterConstraint
+import gg.essential.elementa.constraints.ChildBasedMaxSizeConstraint
+import gg.essential.elementa.constraints.ImageAspectConstraint
+import gg.essential.elementa.constraints.animation.Animations
+import gg.essential.elementa.dsl.*
+import gg.essential.elementa.effects.OutlineEffect
+import gg.essential.elementaPractice.events.WhiteboardEvent
+import gg.essential.elementaPractice.gui.components.ColorToolComponent
+import gg.essential.elementaPractice.gui.components.SliderComponent
+import gg.essential.elementaPractice.gui.components.WhiteboardComponent
+import gg.essential.universal.UMinecraft
+import net.minecraftforge.common.MinecraftForge
+import java.awt.Color
+
+public class WhiteboardGUI : WindowScreen(ElementaVersion.V1) {
+
+    private val whiteboardEvent = WhiteboardEvent(this)
+    private var open = false
+
+    init {
+
+        // TO-DO:
+        // Save/Load buttons?
+
+        // Button to close the whiteboard
+        val closeButton = UIBlock(Color(207, 207, 196)).constrain {
+            x = 5.percent()
+            y = 5.percent()
+            width = ChildBasedMaxSizeConstraint() + 2.pixels()
+            height = ChildBasedMaxSizeConstraint() + 2.pixels()
+        }.onMouseClick {
+            // Close the whiteboard
+            toggle()
+        }.onMouseEnter {
+            // Change color on mouse hover
+            this.animate {
+                setColorAnimation(Animations.OUT_EXP, 0.5f, Color(120, 120, 100).toConstraint())
+            }
+        }.onMouseLeave {
+            // Change color back to original when not hovering
+            this.animate {
+                setColorAnimation(Animations.OUT_EXP, 0.5f, Color(207, 207, 196).toConstraint())
+            }
+        } childOf window
+
+        // Text for close button
+        UIText("X", shadow = false).constrain {
+            x = CenterConstraint()
+            y = CenterConstraint()
+            textScale = 0.5.pixels()
```

Comment:
We try to avoid fractional scales for text because the scaling can be really weird and introduce artifacts. The Minecraft font renderer isn't build to handle scaling for non integets